### PR TITLE
CA-217862: Batch Updates: handle partially upgraded pools

### PR DESCRIPTION
**Pull request by GaborApatiNagy**
Partially upgraded pools and mixed version pools are not supported, XenCenter will display a proper error message.

### CODE REVIEW

File: XenAdmin/Core/Updates.cs [hunk 1]
```diff
@@ -503,7 +491,56 @@ public static UpgradeSequence GetUpgradeSequence(IXenConnection conn, List<XenSe
             }
         }
 
-        private static List<XenServerPatch> GetUpgradeSequenceForHost(Host h, List<XenServerPatch> allPatches, List<XenServerPatch> latestPatches)
+        /// <summary>
+        /// Returns a XenServerVersion if all hosts of the pool has the same version
```
Comment (kc284) [inline XenAdmin/Core/Updates.cs]: "have"

File: XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectServers.cs [hunk 2]
```diff
@@ -193,6 +193,15 @@ private void EnabledRow(Host host, UpdateType type, int index)
                     return;
                 }
 
+                var pool = Helpers.GetPool(host.Connection);
```
Comment (kc284) [inline XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectServers.cs]: The method Updates.GetUpgradeSequence returns null when the host versions are not the same, in which case we'll go into the first if-block and we won't reach down here to show the right reason. This block here needs to move into the first if-block.

**Comment by GaborApatiNagy** [comment]
Thanks @kc284, I have made the necessary code changes.

**Comment by kc284** [comment]
Looks better now. Merging.

**Pull request merged**